sdk/azcore/policy_token_refresh.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcore
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"sync"
+	"time"
+)
+
+const (
+	bearerTokenPrefix = "Bearer "
+)
+
+// DO NOT USE. This is for internal SDK usage. The TokenFetcher is used to provide
+// custom implementations for the bearer token policy.
+type TokenFetcher interface {
+	// IsZeroOrExpired returns a bool if there is a tenant that needs to be initialized
+	IsZeroOrExpired(map[string]time.Time) bool
+	// ShouldRefresh returns a bool if a token needs to be refreshed
+	ShouldRefresh(map[string]time.Time) bool
+	// Fetch performs the GetToken call for the credential and returns the header needed for authentication
+	Fetch(ctx context.Context, cred TokenCredential, opts TokenRequestOptions) (string, error)
+	// Header returns the key for the header in the request
+	Header() string
+}
+
+type tokenRefreshPolicy struct {
+	// cond is used to synchronize token refresh.  the locker
+	// must be locked when updating the following shared state.
+	cond *sync.Cond
+
+	// renewing indicates that the token is in the process of being refreshed
+	renewing bool
+
+	// header contains the authorization header value
+	header string
+
+	// expiresOn is when the token will expire
+	expiresOn map[string]time.Time
```

Comment:
Can't this also be removed and tracked in the implementing types?  We only pass it to `implementer`.

Reply:
Yes done